### DISCUSSION TITLE
curl with Custom Header (certificate in base64)

### DISCUSSION THREAD

**Discussion by Micha1912** [curl/curl#13525] [Q&A]
Hello,

I am trying to build a curl command which gives me a certificate in base64 in a custom header. This header is then read in my application and based on the serial number in the certificate access is granted or not.
Is there such a possibility?

curl --cert certificate.cer:password --key keyfile.key  -H "Content-Type: application/soap+xml; charset=utf-8; action=\"http://url_with_path/"" -H "X-Client-Cert: <complete certificate in base64>" -v --data @"test.xml" https://<webservice> --cacert /etc/cert/ca/root/CA_Cert_1.cer

Best regards,
Michael

**Comment by dfandrich** [comment]
So, you just want to read a header once some kind of curl request is complete? That can be done by adding --write-out '%header{X-My-Header}\n'

**Comment by Micha1912** [comment]
Perhaps I expressed myself badly.
I'm sorry.
I need a certificate in the header (keyword x-forwarded-client-cert (XFCC)) and this certificate in the header is read and used by the application in the header as a base64 string.
I need the whole thing to recreate a test via curl.

Normally the request goes from the client to the reverse proxy and this reads the client certificate and then appends it to the header and the application can then read it.

I am now trying to take the reverse proxy out of the equation and send the request directly to the application. To do this, the certificate must be included in the header.

**Comment by dfandrich** [comment]
So, you just want to *send* a header? That's done with
-H "x-forwarded-client-cert: <contents of header here>". The contents can be
substituted with the contents of a file or filter (like base64) using your
shell. Or maybe I'm still not following.

**Reply by bagder** [reply]
You can also substitute it with a file (even base64 it) using [command line variables](https://everything.curl.dev/cmdline/variables.html) since curl 8.3.0.
